### ISSUE TITLE
Optuna Picks trial 0 nearly every time including the same params

### ISSUE THREAD

**Issue by rm17tink** [optuna/optuna#3479]
### Environment

Environment
Environment
Optuna version: 2.10.0
Python version: 3.6
OS: Windows 10
(Optional) Other libraries and their versions: lightgbm = 3.3.2

### Description

I have been running quite a few Optuna Model with Lightgbm and CV. When I run many trials....i almost always get the first trail as being the best study(maximizing avg[valid_auc]). From what i know of Baysian search shouldnt the model find small improvements given the scope of parameters I am passing to it. it seems like no matter what columns I pass and or ratio of training the same parameters are picked...but picked immediately(not learned).  

### Error messages, stack traces, or logs

```shell
No error messages.
```

### Steps to reproduce

```python

def objective_no_cv(trial, X, y):
    
    global model
    
    cv = StratifiedKFold(n_splits=5, shuffle=True, random_state=1121218)

    cv_scores = np.empty(5)
    for idx, (train_idx, valid_idx) in enumerate(cv.split(X, y)):
        X_train, X_valid = X.iloc[train_idx], X.iloc[valid_idx]
        y_train, y_valid = y[train_idx], y[valid_idx]
    
    #CV storage value
    auc_scores_train   = np.empty(5)
    auc_scores_valid   = np.empty(5)
    valid_train_spread = np.empty(5)
    
    param_grid = {
        #         "device_type": trial.suggest_categorical("device_type", ['gpu']),
        "n_estimators": trial.suggest_categorical("n_estimators", [10000]),
        "verbose": -1,  # mutes the [WARNING] output
        "boosting_type": "gbdt", #dart
         "seed": 42,
        #"scale_pos_weight": 583, #(makes performance terrible)
        #"is_unbalance": "true",  #added ( makes performance terrible)
        "learning_rate": trial.suggest_float("learning_rate", 0.1, 0.3),
        "num_leaves": trial.suggest_int("num_leaves", 20, 3000, step=20),
        "max_depth": trial.suggest_int("max_depth", 3, 12),
        "min_data_in_leaf": trial.suggest_int("min_data_in_leaf", 200, 10000, step=100),
        "max_bin": trial.suggest_int("max_bin", 200, 300),
        "lambda_l1": trial.suggest_int("lambda_l1", 0, 100, step=5),
        "lambda_l2": trial.suggest_int("lambda_l2", 0, 100, step=5),
        "min_gain_to_split": trial.suggest_float("min_gain_to_split", 0, 15),
        "bagging_fraction": trial.suggest_float(
            "bagging_fraction", 0.2, 0.95, step=0.1
        ),
        "bagging_freq": trial.suggest_categorical("bagging_freq", [1]),
        "feature_fraction": trial.suggest_float(
            "feature_fraction", 0.2, 0.95, step=0.1
        ),
    }
    model = lgb.LGBMClassifier(objective="binary", **param_grid)
    model.fit(
            X_train,
            y_train,
            eval_set=[(X_valid, y_valid)],
            eval_metric="auc",
            early_stopping_rounds=100,
            callbacks=[
                LightGBMPruningCallback(trial, "auc")
            ],  # Add a pruning callback
        )
   
    # TRAINING INFO auc.
    train_preds = model.predict_proba(X_train)
    auc_scores_train[idx] = roc_auc_score(y_train, train_preds[:,1])
    trial.set_user_attr("auc_scores_train", np.mean(auc_scores_train))    
    
    
    #VALID INFO (measurement we maximize in the trials)
    valid_preds = model.predict_proba(X_valid)
    auc_scores_valid[idx] = roc_auc_score(y_valid, valid_preds[:,1])
    
    
    #Valid Train Spread 
    valid_train_spread[idx] = (auc_scores_train - auc_scores_valid)
    trial.set_user_attr("valid_train_spread", np.mean(valid_train_spread))
    
    return np.mean(auc_scores_valid)

if __name__ == "__main__":
    study = optuna.create_study(pruner=optuna.pruners.MedianPruner(n_warmup_steps=10)
                                , direction="maximize", study_name="LGBM Classifier"
                                ,sampler=TPESampler(seed=42))
    #supress logging warnings
    #optuna.logging.set_verbosity(optuna.logging.WARNING)
    func = lambda trial: objective_no_cv(trial 
                                         # pass in the datasets
                                         ,X_train ,y_train
                                         ,X_valid ,y_valid
                                        )
    study.optimize(func, n_trials=10, callbacks=[callback])
```

**Comment by nzw0301** [comment]
Hi, could you share minimal reproducible codes with us?

**Comment by rm17tink** [comment]
Hi. Im sorry is that not reproducible? I know this might be an abstract question and a misunderstanding of using Lightgbm with Optuna but should the study constantly provide better results with each parmas iteration?

**Comment by nzw0301** [comment]
> is that not reproducible? 

Since we cannot run the code without error due to missing import statements and data definition, I'm afraid that the code looks irreproducible. If the code is irreproducible, we really find it difficult to investigate the issue. As a result, no one can answer this question.

**Comment by rm17tink** [comment]
Can we confirm that the above design still implements the Bayesian approach to tuning? What is a good standard set of trials?

**Comment by nzw0301** [comment]
If you mean TPE as a Bayesian approach, the answer is no because the first 10 trials use random search, not TPE. The documentation page says

> `n_startup_trials` – The random sampling is used instead of the TPE algorithm until the given number of trials finish in the same study.

**Comment by rm17tink** [comment]
I am not passing in `n_startup_trials` just a seed.

**Comment by nzw0301** [comment]
Well, the default value is `n_startup_trials=10` defined as https://github.com/optuna/optuna/blob/1a7a51b23af63cc053f4c98fa6c02e599fa21d79/optuna/samplers/_tpe/sampler.py#L220...

**Comment by rm17tink** [comment]
So TPE is not TPE until after 10?

**Comment by rm17tink** [comment]
Alright thanks. So I would have to adjust the default n_startup_trials to `sampler=TPESampler(seed=42, n_startup_trials = 20)`

**Comment by nzw0301** [comment]
Well, it depends on the problem; it might be worth doing that.
